### PR TITLE
Add `allow-all-egress` policy for `cni: cilium`

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/20946

### CODE REVIEW

File: helm/cluster-resources/values.yaml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # TEMPLATE-APP: This is set as a reasonable default, feel free to change.
 
 name: cluster-resources
-namespace: kube-system
```
Comment (glitchcrab) [inline helm/cluster-resources/values.yaml:4]: Is this defaulted elsewhere?

**Review by glitchcrab — COMMENTED** [review]
👌

**Review by glitchcrab — APPROVED** [review]

File: helm/cluster-resources/templates/_helpers.tpl [hunk 2]
```diff
@@ -8,6 +8,10 @@ Expand the name of the chart.
 {{- .Chart.Name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{- define "namespace" -}}
```
Comment (tfussell) [inline helm/cluster-resources/templates/_helpers.tpl]: This doesn't seem to be being used.

File: helm/cluster-resources/values.yaml (hunk 1 above)
Comment (kopiczko) [reply helm/cluster-resources/values.yaml:4]: I don't see this being practical. https://github.com/giantswarm/template-app/pull/29

File: helm/cluster-resources/templates/_helpers.tpl (hunk 2 above)
Comment (kopiczko) [reply helm/cluster-resources/templates/_helpers.tpl]: Removed.

**Review by tfussell — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-- Add NetworkPolicy for `coredns`.
+- Add `allow-all-egress` CiliumClusterwideNEtworkPolicy for `cni: cilium`.
```
Comment (tfussell) [inline CHANGELOG.md]:
```suggestion
- Add `allow-all-egress` CiliumClusterwideNetworkPolicy for `cni: cilium`.
```

File: helm/cluster-resources/values.yaml [hunk 4]
```diff
@@ -13,3 +12,5 @@ image:
   name: giantswarm/cluster-resources
   tag: v0.0.1
   pullPolicy: IfNotPresent
+
+cni: "cilium"
```
Comment (tfussell) [inline helm/cluster-resources/values.yaml]:
```suggestion
cni: cilium
```

**Review by tfussell — APPROVED** [review]
Two superficial suggestions. Otherwise LGTM.

**Pull request merged**